### DISCUSSION TITLE
Game loop runs slower on another computer

### DISCUSSION THREAD

**Discussion by TeodorDyakov** [hajimehoshi/ebiten#1560] [Q&A]
Hello. I have recently made a game with Ebiten. The game was made on my Ubuntu workstation. When I run the game it runs smoothly and fast(the machine has i3 10th gen CPU and runs Ubuntu). I cross compiled the game for my other laptop (Windows 10, i7 CPU), and tried running the game on it. Unfortunately The game appeared noticeably slower. The animation of a falling object takes roughly two times more.  I wonder if this problem is with cross compilation or just that my other latpopt CPU is too slow?

**Comment by hajimehoshi** [comment]
Hi,

Perhaps did you write your game logic in `Draw` instead of `Update`? `Draw`'s frequency depends on environment, especially your display's refresh rate.

**Comment by SolarLune** [comment]
Yeah, it could be a few things - it would be worth it to post your code if possible. It could be something related to the rendering; perhaps it could even be a Vsync-related issue.

**Reply by TeodorDyakov** [reply]
The code is pretty long https://github.com/TeodorDyakov/spooky-connect4/blob/main/client/graphics/guiGame.go
Generally I have only calls to draw text or images in the draw function.

**Reply by TeodorDyakov** [reply]
I see now that I update the coordinates of the falling marbles in the draw function. Perhpas thats the problem.
